loadtest/user/userentity/websocket.go

```diff
@@ -139,9 +139,10 @@ func (ue *UserEntity) wsEventHandler(ev *model.WebSocketEvent) error {
 // Only on calling Disconnect explicitly, it will return.
 func (ue *UserEntity) listen(errChan chan error) {
 	connectionFailCount := 0
+	var err error
 start:
 	for {
-		client, err := websocket.NewClient4(&websocket.ClientParams{
+		ue.wsClient, err = websocket.NewClient4(&websocket.ClientParams{
```

Comment:
I realize this is breaking from our established norm. However, it's been a long time since anyone touched this code and I don't think there is anything inherently bad by storing the client in the struct. I know the actual client is pretty resilient against failures, so I don't foresee any issue calling its methods from outside the main goroutine.

To push all activity inside the main for-select loop would require further channels and more coordination which didn't sit right with me. The only edge-case is that the call will fail when the client is closing, which I think is fine. LMK.

Reply:
The reason we never did that is because it's inherently racy and we didn't want the complexity of adding a mutex to `UserEntity`. Now any of the `UpdateXXX` methods has a race condition in accessing the client object.

One decent way to solve this would be to repurpose the `wsTyping` channel to be a generic event channel and then switch on event type to call the appropriate method.

Reply:
No other concerns other than the race accessing the client object which is the reason we kept it internal to `listen` but I see you took care of that so all good :+1: